### PR TITLE
- Remove Save method from GenericRepository

### PR DESCRIPTION
**Pull request by Finka95**
- Changed TEntity parameter in Update to params Object[]
- Added Global Tracking as NoTracking
- Changed some strings parameters to Constant

### CODE REVIEW

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public async Task<Ticket> GetTicketById(int id, CancellationToken cancellationTo
 
     public async Task<IEnumerable<Ticket>> GetTickets(CancellationToken cancellationToken)
     {
-        var ticketsEntity = await _ticketRepository.GetWithInclude(cancellationToken, false, null, null,
+        var ticketsEntity = await _ticketRepository.GetWithInclude(cancellationToken, null, null,
```
Comment (nantonov) [inline TicketSystem/TicketSystem.BLL/Services/TicketService.cs:52]: Can we make null values as optional parameters in the method?

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs [hunk 2]
```diff
@@ -62,7 +62,7 @@ public async Task<Ticket> UpdateTicket(Ticket ticket, CancellationToken cancella
     {
         var ticketEntity = _mapper.Map<TicketEntity>(ticket);
 
-        await _ticketRepository.Update(ticketEntity, cancellationToken);
+        await _ticketRepository.Update(cancellationToken, ticketEntity);
```
Comment (nantonov) [inline TicketSystem/TicketSystem.BLL/Services/TicketService.cs:65]: Looks like the order of the parameters was right :)

**Review by nantonov — CHANGES_REQUESTED** [review]

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs (hunk 2 above)
Comment (v12code) [reply TicketSystem/TicketSystem.BLL/Services/TicketService.cs:65]: I expanded this method by adding params TEntity[]

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs (hunk 1 above)
Comment (v12code) [reply TicketSystem/TicketSystem.BLL/Services/TicketService.cs:52]: image
these are optional
but at the end there are params includeProperties, so it is necessary to specify them when calling the method

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs (hunk 2 above)
Comment (nantonov) [reply TicketSystem/TicketSystem.BLL/Services/TicketService.cs:65]: Looks ugly, but okay
I suggest to separate `Update `and `UpdateRange `methods in the future

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs (hunk 1 above)
Comment (nantonov) [reply TicketSystem/TicketSystem.BLL/Services/TicketService.cs:52]: Okay, it's super generic :D

**Review by nantonov — APPROVED** [review]

**Review by Genresal — APPROVED** [review]

**Review by 5Ka-me — APPROVED** [review]

**Pull request merged**